docsite/docs/Docs/fal-cli/environments.md

```diff
@@ -0,0 +1,294 @@
+# Isolated scripts (and environment management)
+
+If you have Python models (or Fal hooks) running up in your project with
+dependencies against external packages (like `scikit-learn` for machine learning
+models or `slack-sdk` for alerts ) you can let Fal manage them for you without
+even thinking they are there. This provides extensive interoperability when
+sharing this project with others (like your co-workers) and ensuring that the
+local runs you are making are easily reproducible in different environments
+(like in your CI/CD process).
+
+> Note: Isolated scripts feature is still experimental, and there might be both
+> internal and external (user-visible) changes on how these environments are
+> defined and managed.
+
+## Defining environments
+
+All the different environments that your Fal project might need can be defined
+inside `fal_project.yml`, a file that is located in the same directory as your
+`dbt_project.yml`, under your project's root directory.
+
+An example `fal_project.yml` might look like this, where you have a list of
+named environments:
+
+```yml
+environments:
+  - name: alerts
+    type: venv
+    requirements:
+      - slack-sdk==1.1.0
+
+  - name: training
+    type: conda
+    packages:
+      - scikit-learn=2.0.0
+```
+
+Each environment comes with a unique name (an identifier for you to reference it
+later on, inside your `schema.yml`), a type (which would allow you to choose the
+environment management backend that you want to use) and a list of configuration
+options for the selected environment manager.
+
+### Using virtual environments (for `pip` packages)
+
+If your hook depends on packages from [PyPI](https://pypi.org/), you can create
+an environment with the type `venv` and Fal will take care of creating and
+managing (e.g. installing all the required packages) that virtual environment
+for you. The only parameter for `venv` based installations is the `requirements`
+section, which is just a list of PyPI packages (and their pinned versions) that
+your script needs.
```

Comment:
Maybe we can restructure the first sentence there. Just breaking it in two would make it easier to parse.